metrics/otelgrpc/otelgrpc.go

```diff
@@ -0,0 +1,159 @@
+package otelgrpc
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/goto/meteor/utils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+type UnaryParams struct {
+	Start  time.Time
+	Method string
+	Req    any
+	Res    any
+	Err    error
+}
+
+type Monitor struct {
+	duration     metric.Int64Histogram
+	requestSize  metric.Int64Histogram
+	responseSize metric.Int64Histogram
+	attributes   []attribute.KeyValue
+}
+
+func NewOtelGRPCMonitor(hostName string) Monitor {
+	meter := otel.Meter("github.com/goto/meteor/metrics/otelgrpc")
+
+	duration, err := meter.Int64Histogram("rpc.client.duration", metric.WithUnit("ms"))
+	if err != nil {
+		handleOtelErr(err)
+	}
```

Comment:
```suggestion
	handleOtelErr(err)
```

similarly for the other error checks.